php/api/endpoint/class-report.php

```diff
@@ -0,0 +1,145 @@
+<?php
+/**
+ * This file is responsible for creating the /report/ API endpoint..
+ *
+ * @package WP_Tide_API
+ */
+
+namespace WP_Tide_API\API\Endpoint;
+
+use WP_Tide_API\Base;
+use WP_Tide_API\Utility\User;
+
+/**
+ * Class Report
+ */
+class Report extends Base {
+
+	/**
+	 * API namespace.
+	 *
+	 * @var string The namespace.
+	 */
+	private $namespace = 'tide/v1';
+
+	/**
+	 * Base path for reports.
+	 *
+	 * @var string Base path.
+	 */
+	private $rest_base = 'report';
+
+	const POST_PATTERN     = '(?P<post_id>\d*)';
+	const CHECKSUM_PATTERN = '(?P<checksum>[a-fA-F\d]{64})';
+	const STANDARD_PATTERN = '(?P<standard>[\w-]+)';
+
+
+	/**
+	 * Register new routes.
+	 *
+	 * @action rest_api_init
+	 */
+	public function register_routes() {
+
+		register_rest_route( $this->namespace, '/' . $this->rest_base, array(
+			'methods'  => 'GET',
+			'callback' => array( $this, 'report_response' ),
+		) );
+
+		register_rest_route( $this->namespace, '/' . $this->rest_base . '/' . static::CHECKSUM_PATTERN . '/' . static::STANDARD_PATTERN, array(
+			'methods'  => 'GET',
+			'callback' => array( $this, 'report_response' ),
+		) );
+
+		register_rest_route( $this->namespace, '/' . $this->rest_base . '/' . static::POST_PATTERN . '/' . static::STANDARD_PATTERN, array(
+			'methods'  => 'GET',
+			'callback' => array( $this, 'report_response' ),
+		) );
+	}
+
+	/**
+	 * Prepare the response for a report requests.
+	 *
+	 * @param \WP_REST_Request $request The request object.
+	 *
+	 * @return mixed|\WP_REST_Response The response object.
+	 */
+	public function report_response( \WP_REST_Request $request ) {
+
+		// These reports are not for unauthenticated users.
+		if ( ! User::authenticated() ) {
+			return rest_ensure_response( $this->report_error( 'unauthenticated_call', __( 'unauthenticated report request', 'tide-api' ), 301 ) );
+		}
+
+		// @todo get the cloud storage source from configuration. For now this is AWS S3 only.
+		$object_source = 'aws_s3';
```

Comment:
Note:  This will come from the `$meta['full']['type']` key and needs to be aligned with `$plugin->components[TYPE]`.